addons/html_editor/static/src/main/toolbar/toolbar_plugin.js

```diff
@@ -309,17 +309,17 @@ export class ToolbarPlugin extends Plugin {
     }
 
     /**
-     * @returns {Object<string, ToolbarButton[]>}
+     * @returns ToolbarButton[]
      */
-    getButtonsByNamespace() {
-        const namespaces = this.getResource("toolbar_namespaces").map((ns) => ns.id);
-        const buttonsByNamespace = {};
-        for (const namespace of namespaces) {
-            buttonsByNamespace[namespace] = this.buttonGroups.flatMap((group) =>
-                group.buttons.filter((btn) => btn.namespaces.includes(namespace))
-            );
+    getButtonsForNamespace(namespace) {
+        if (this.buttonsByNamespace[namespace]) {
+            return this.buttonsByNamespace[namespace];
         }
-        return buttonsByNamespace;
+        const button = this.buttonGroups.flatMap((group) =>
+            group.buttons.filter((btn) => btn.namespaces.includes(namespace))
+        );
+        this.buttonsByNamespace[namespace] = button;
```

Comment:
I feel like this supposed to be some kind of caching mecanism but, if I'm not mistaken, the whole object is going to be recreated each time I query a non-existent namespace, correct  ?